classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.php

```diff
@@ -112,9 +112,11 @@ public function up(): void
             ->leftJoin('queries AS q', 'n.assoc_id', '=', 'q.query_id')
             ->where('n.assoc_type', '=', self::ASSOC_TYPE_QUERY)
             ->whereNull('q.query_id')
-            ->pluck('n.note_id', 'n.assoc_id');
+            ->select('n.note_id', 'n.assoc_id')
+            ->lazyById(1000, 'n.note_id', 'note_id');
```

Comment:
I've just added this `lazyById()`, but this migration can still be highly optimized, as there's a couple of SQL commands running inside the loops.